ironic/provider.go

```diff
@@ -20,13 +20,14 @@ func Provider() terraform.ResourceProvider {
 			"microversion": {
 				Type:        schema.TypeString,
 				Required:    true,
```

Comment:
This makes an API SIG member in me really, really sad.

Reply:
Could you elaborate? I’m happy to change it but I don't know what the proper thing to do is. This is the same approach as gopher cloud.

Reply:
The gopher cloud approach is not ideal, really.. I did not want to teach them how to manage their project, but microversions are not supposed to be exposed to users like that. It's largely our fault, as OpenStack API SIG, to fail to communicate it to people well enough. Please check my specifications on how we see it done: http://specs.openstack.org/openstack/api-sig/guidelines/sdk-exposing-microversions.html